src/PowerShellEditorServices/Extensions/EditorFileRanges.cs

```diff
@@ -0,0 +1,349 @@
+        /// </summary>
+        IFilePosition End { get; }
+    }
+
+    /// <summary>
+    /// A snapshot of a file, including the URI of the file
+    /// and its textual contents when accessed.
+    /// </summary>
+    public interface IFileContext
+    {
+        /// <summary>
+        /// The URI of the file.
+        /// </summary>
+        Uri Uri { get; }
+
+        /// <summary>
+        /// The content of the file when it was accessed.
+        /// </summary>
+        string Content { get; }
+    }
+
+    /// <summary>
+    /// 0-based position within a file, conformant with the Language Server Protocol.
+    /// </summary>
+    public interface ILspFilePosition
+    {
+        /// <summary>
+        /// The line index of the position within the file.
+        /// </summary>
+        long Line { get; }
+
+        /// <summary>
+        /// The character offset from the line of the position.
+        /// </summary>
+        long Character { get; }
+    }
+
+    /// <summary>
+    /// 0-based range within a file, conformant with the Language Server Protocol.
+    /// </summary>
+    public interface ILspFileRange
+    {
+        /// <summary>
+        /// The start position of the range.
+        /// </summary>
+        ILspFilePosition Start { get; }
+
+        /// <summary>
+        /// The end position of the range.
+        /// </summary>
+        ILspFilePosition End { get; }
+    }
+
+    /// <summary>
+    /// Snapshot of a file in focus in the editor.
+    /// </summary>
+    public interface ILspCurrentFileContext : IFileContext
+    {
+        /// <summary>
+        /// The language the editor associates with this file.
+        /// </summary>
+        string Language { get; }
+
+        /// <summary>
+        /// The position of the cursor within the file when it was accessed.
+        /// If the cursor is not in the file, values may be negative.
+        /// </summary>
+        ILspFilePosition CursorPosition { get; }
+
+        /// <summary>
+        /// The currently selected range when the file was accessed.
+        /// If no selection is made, values may be negative.
+        /// </summary>
+        ILspFileRange SelectionRange { get; }
+    }
+
+    internal struct OmnisharpLspPosition : ILspFilePosition
+    {
+        private readonly Position _position;
+
+        public OmnisharpLspPosition(Position position)
+        {
+            _position = position;
+        }
+
+        public long Line => _position.Line;
+
+        public long Character => _position.Character;
+    }
+
+    internal struct OmnisharpLspRange : ILspFileRange
+    {
+        private readonly Range _range;
+
+        public OmnisharpLspRange(Range range)
+        {
+            _range = range;
+        }
+
+        public ILspFilePosition Start => new OmnisharpLspPosition(_range.Start);
+
+        public ILspFilePosition End => new OmnisharpLspPosition(_range.End);
+    }
+
+    internal struct BufferFilePosition : IFilePosition
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IEquatable<T>' in value type 'BufferFilePosition'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4979332)